commands/scale_cluster.go

```diff
@@ -38,115 +39,307 @@ Examples:
+// defaultScaleClusterArguments defaults arguments supplied by the users.
+func defaultScaleClusterArguments(cmd *cobra.Command, clusterId string, maxBefore int64, minBefore int64) scaleClusterArguments {
 	endpoint := config.Config.ChooseEndpoint(cmdAPIEndpoint)
 	token := config.Config.ChooseToken(endpoint, cmdToken)
 	scheme := config.Config.ChooseScheme(endpoint, cmdToken)
 
-	return scaleClusterArguments{
+	scaleArgs := scaleClusterArguments{
 		apiEndpoint:         endpoint,
 		authToken:           token,
-		scheme:              scheme,
-		clusterID:           cmdClusterID,
+		clusterID:           clusterId,
 		numWorkersDesired:   cmdNumWorkers,
-		workerNumCPUs:       cmdWorkerNumCPUs,
-		workerMemorySizeGB:  cmdWorkerMemorySizeGB,
-		workerStorageSizeGB: cmdWorkerStorageSizeGB,
 		oppressConfirmation: cmdForce,
+		scheme:              scheme,
 		verbose:             cmdVerbose,
+		workersMax:          cmdWorkersMax,
+		workersMin:          cmdWorkersMin,
+	}
+
+	if !cmd.Flags().Changed(cmdWorkersMinName) {
+		scaleArgs.workersMin = minBefore
+	}
+	if !cmd.Flags().Changed(cmdWorkersMaxName) {
+		scaleArgs.workersMax = maxBefore
+	}
+	if !cmd.Flags().Changed(cmdWorkersMaxName) && !cmd.Flags().Changed(cmdWorkersMinName) && cmd.Flags().Changed(cmdWorkersNumName) {
+		scaleArgs.workersMax = int64(scaleArgs.numWorkersDesired)
+		scaleArgs.workersMin = int64(scaleArgs.numWorkersDesired)
 	}
-}
 
-type scaleClusterResults struct {
-	// number of workers according to our info, just before the PATCH call.
-	numWorkersBefore int
-	// number of workers to add, just before the PATCH call. Might be negative.
-	numWorkersToAdd int
-	// number of workers as of the PATCH call response.
-	numWorkersAfter int
+	return scaleArgs
 }
 
-func init() {
-	ScaleClusterCommand.Flags().BoolVarP(&cmdForce, "force", "", false, "If set, no confirmation is required when reducing the number of workers.")
-	ScaleClusterCommand.Flags().IntVarP(&cmdNumWorkers, "num-workers", "w", 0, "Number of worker nodes to have after scaling.")
-	ScaleClusterCommand.Flags().IntVarP(&cmdWorkerNumCPUs, "num-cpus", "", 0, "Number of CPU cores per added worker node.")
-	ScaleClusterCommand.Flags().Float32VarP(&cmdWorkerMemorySizeGB, "memory-gb", "", 0, "RAM per added worker node.")
-	ScaleClusterCommand.Flags().Float32VarP(&cmdWorkerStorageSizeGB, "storage-gb", "", 0, "Local storage size per added worker node.")
+// getClusterStatus returns the status for one cluster.
+func getClusterStatus(clusterID, activityName string) (*v1alpha1.StatusCluster, error) {
+	// perform API call
+	auxParams := ClientV2.DefaultAuxiliaryParams()
+	auxParams.ActivityName = activityName
 
-	ScaleCommand.AddCommand(ScaleClusterCommand)
-}
+	response, err := ClientV2.GetClusterStatus(clusterID, auxParams)
+	if err != nil {
+		if IsAPIError(err) {
+			clientErr, ok := err.(*clienterror.APIError)
+			if ok {
+				switch clientErr.HTTPStatusCode {
+				case http.StatusForbidden:
```

Comment:
I would actually use matchers for all these cases as well to just have the flat error checks hierarchy as suggested below. Without all the conditional indentions. We could have 1 instead of 4 indentation levels. 

```
if IsStatusForbiddenError(err) {
  ...
} else if IsStatusUnauthorized(err) {
  ...
} else if IsStatusNotFound(err) {
  ...
} else if IsStatusInternalServerError(err) {
  ...
} else if err != nil {
  ...
}
```